lib/api/objectDelete.js

```diff
@@ -54,8 +56,21 @@ export default function objectDelete(authInfo, request, log, cb) {
                     contentLength: objMD['content-length'],
                 });
             }
-            return services.deleteObject(bucketName, objMD, objectKey, log,
-                err => {
+            const objectLocationConstraint = request
+                .headers[constants.objectLocationConstraintHeader];
+            const bucketLocationConstraint = bucket.getLocationConstraint();
+            const requestEndpoint = request.parsedHost;
+            if (!BackendInfo.areValidBackendParameters(objectLocationConstraint,
+                bucketLocationConstraint, requestEndpoint, log)) {
+                return process.nextTick(() => {
+                    cb(errors.InvalidArgument
+                        .customizeDescription('Location Constraint Info is ' +
+                        'invalid.'));
+                });
+            }
```

Comment:
cool.  Not sure why it came up in my review.  Github had a historical moment.  Other than the cors stuff, this PR is looking good to me 😃